lib/galaxy/jobs/runners/kubernetes.py

```diff
@@ -136,15 +145,17 @@ def queue_job(self, job_wrapper):
             k8s_job_prefix,
             self.__get_k8s_job_spec(ajs)
         )
-
         job = Job(self._pykube_api, k8s_job_obj)
-        job.create()
-        job_id = job.metadata['name']
+        try:
+            job.create()
+            job_id = job.metadata.get('name')
+            # define job attributes in the AsyncronousJobState for follow-up
+            ajs.job_id = job_id
+            # store runner information for tracking if Galaxy restarts
+            job_wrapper.set_external_id(job_id)
+        except Exception as e:
+            log.debug(str(e))
```

Comment:
You probably want to re-raise the exception right?

Reply:
Done in https://github.com/galaxyproject/galaxy/pull/11383/commits/7bfa28dfa87952e941bbb4aec77c6ef63d2a2eb8